raiden-ts/src/services/epics/udc.ts

```diff
@@ -280,42 +281,60 @@ export function udcWithdrawPlanRequestEpic(
 }
 
 /**
- * At startup, check if there was a previous plan and re-emit udcWithdrawPlan.success action
+ * If config.autoUdcWithdraw is enabled, monitors planned withdraws and udcWithdraw.request when
+ * ready
  *
  * @param action$ - Observable of RaidenActions
  * @param state$ - Observable of RaidenStates
  * @param deps - Epics dependencies
  * @param deps.userDepositContract - UDC contract instance
  * @param deps.address - Our address
  * @param deps.config$ - Config observable
+ * @param deps.log - Logger instance
  * @returns Observable of udcWithdrawPlan.success actions
  */
-export function udcCheckWithdrawPlannedEpic(
-  {}: Observable<RaidenAction>,
+export function udcAutoWithdrawEpic(
+  action$: Observable<RaidenAction>,
   {}: Observable<RaidenState>,
-  { userDepositContract, address, config$ }: RaidenEpicDeps,
-): Observable<udcWithdrawPlan.success> {
-  return config$.pipe(
-    first(),
-    mergeMap(({ pollingInterval }) =>
-      retryAsync$(() => userDepositContract.withdraw_plans(address), pollingInterval, {
-        onErrors: networkErrors,
-      }),
-    ),
-    filter((value) => value.withdraw_block.gt(Zero)),
-    map(({ amount, withdraw_block }) =>
-      udcWithdrawPlan.success(
-        { block: withdraw_block.toNumber(), confirmed: true },
-        { amount: amount as UInt<32> },
+  { userDepositContract, address, config$, log }: RaidenEpicDeps,
+): Observable<udcWithdraw.request> {
+  const WITHDRAW_TIMEOUT = 100;
```

Comment:
Random line of this function...
I'm not too satisfied by this approach with the blocks and the timeout. Why can't we simply listen for the withdraw plan success actions as trigger? There we can also read-out the blocks to wait. And if I'm not mistaken these actions are re-emitted after each restart, aren't they?

Reply:
We had a call to this topic. With his good explanation this algorithm satisfies me.

Reply:
I'll move this constant to the `constants.ts` file and remove the docstrings references.

Reply:
:heavy_check_mark: